src/org/rascalmpl/uri/URIResolverRegistry.java

```diff
@@ -263,75 +320,84 @@ public ISourceLocation logicalToPhysical(ISourceLocation loc) throws IOException
+					loc = backup.resolve(URIUtil.removeOffset(prev));
+					removedOffset = true;
+				}
 
-					if (loc != null && !prev.equals(loc)) {
-						break; // continue to offset/length handling below with found location
-					}
+				if (loc != null && !prev.equals(loc)) {
+					break; // continue to offset/length handling below with found location
 				}
 			}
+		}
 
-			if (loc == null || prev.equals(loc)) {
-				return null;
-			}
+		if (loc == null || prev.equals(loc)) {
+			return null;
+		}
 
-			if (removedOffset || !loc.hasOffsetLength()) { // then copy the offset from the logical one
-				if (prev.hasLineColumn()) {
-					loc = vf.sourceLocation(loc, prev.getOffset(), prev.getLength(), prev.getBeginLine(),
-						prev.getEndLine(), prev.getBeginColumn(), prev.getEndColumn());
-				}
-				else if (prev.hasOffsetLength()) {
-					if (loc.hasOffsetLength()) {
-						loc = vf.sourceLocation(loc, prev.getOffset() + loc.getOffset(), prev.getLength());
-					}
-					else {
-						loc = vf.sourceLocation(loc, prev.getOffset(), prev.getLength());
-					}
-				}
+		if (removedOffset || !loc.hasOffsetLength()) { // then copy the offset from the logical one
+			if (prev.hasLineColumn()) {
+				return vf.sourceLocation(loc, prev.getOffset(), prev.getLength(), prev.getBeginLine(),
+					prev.getEndLine(), prev.getBeginColumn(), prev.getEndColumn());
 			}
-			else if (loc.hasLineColumn()) { // the logical location offsets relative to the physical offset, possibly
-											// including line numbers
-				if (prev.hasLineColumn()) {
-					loc = vf.sourceLocation(loc, loc.getOffset() + prev.getOffset(), loc.getLength(),
-						loc.getBeginLine() + prev.getBeginLine() - 1, loc.getEndLine() + prev.getEndLine() - 1,
-						loc.getBeginColumn(), loc.getEndColumn());
+			else if (prev.hasOffsetLength()) {
+				if (loc.hasOffsetLength()) {
+					return vf.sourceLocation(loc, prev.getOffset() + loc.getOffset(), prev.getLength());
 				}
-				else if (prev.hasOffsetLength()) {
-					loc = vf.sourceLocation(loc, loc.getOffset() + prev.getOffset(), loc.getLength());
+				else {
+					return vf.sourceLocation(loc, prev.getOffset(), prev.getLength());
 				}
 			}
-			else if (loc.hasOffsetLength()) { // the logical location offsets relative to the physical one
-				if (prev.hasOffsetLength()) {
-					loc = vf.sourceLocation(loc, loc.getOffset() + prev.getOffset(), loc.getLength());
-				}
+		}
+		else if (loc.hasLineColumn()) { // the logical location offsets relative to the physical offset, possibly
+										// including line numbers
+			if (prev.hasLineColumn()) {
+				return vf.sourceLocation(loc, loc.getOffset() + prev.getOffset(), loc.getLength(),
+					loc.getBeginLine() + prev.getBeginLine() - 1, loc.getEndLine() + prev.getEndLine() - 1,
+					loc.getBeginColumn(), loc.getEndColumn());
+			}
+			else if (prev.hasOffsetLength()) {
+				return vf.sourceLocation(loc, loc.getOffset() + prev.getOffset(), loc.getLength());
```

Comment:
I think but I'm not sure that most of the above changes were whitespace/indentation. It's hard to see. If you say you ran a code format/fix indent command, then I'm ok with that.

Reply:
Yup, I moved them to a separate method, and therefore they shifted up an indentation.